File: app/src/main/java/net/daverix/ajvm/jvm/IntegerObject.java
package net.daverix.ajvm.jvm;


import net.daverix.ajvm.VirtualObject;

import java.io.IOException;

public class IntegerObject implements VirtualObject {
    private Integer integer;

    public IntegerObject() {
    }

    @Override
    public void initialize(Object[] args) {
        integer = (Integer) args[0];
    }

    @Override
    public String getName() {
        return "java/lang/Integer";
    }

    @Override
    public void setFieldValue(String fieldName, Object value) {

    }

    @Override
    public Object getFieldValue(String fieldName) {
        return null;
    }

    @Override
    public Object invokeMethod(String name, String descriptor, Object... args) throws IOException {
        if("parseInt".equals(name) && descriptor.equals("(Ljava/lang/String;)I")) {
            return Integer.parseInt((String) args[0]);
        }

        return null;
    }
}

File: app/src/main/java/net/daverix/ajvm/IntegerObject.kt
/*
    Java Virtual Machine for Android
    Copyright (C) 2017 David Laurell

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
 */
package net.daverix.ajvm

import java.io.IOException

class IntegerObject : VirtualObject {
    private var integer: Int? = null

    override val name: String
        get() = "java/lang/Integer"

    override fun initialize(args: Array<Any>) {
        integer = args[0] as Int
    }

    override fun setFieldValue(fieldName: String, value: Any?) {

    }

    override fun getFieldValue(fieldName: String): Any? {
        return null
    }

    @Throws(IOException::class)
    override fun invokeMethod(name: String, descriptor: String, args: Array<Any?>): Any? {
        return if ("parseInt" == name && descriptor == "(Ljava/lang/String;)I") {
            Integer.parseInt(args[0] as String)
        } else null

    }
}
